### ISSUE TITLE
Google, we have a problem

### ISSUE THREAD

**Issue by charliedotau** [opennem/opennem-fe#5]
This one is about improving how well your site's potential search engine ranking.

## Some Observations

These observations are of the live site (not the development site).

1. the main page of the App has little to no copy (e.g words).  Google really doesn't like this (part of its algorithm looks at the ratio of content to code).  If you look at the page source of the main page you'll see what I mean.  It's almost entirely HTML markup.  Very few words.

2. the markup contains no visible link to the `/#/about` page.   This suggests (I might be wrong) that the link to the page is added with javascript.

3. whilst Google has [clearly indexed the site](https://www.google.com.au/search?q=site%3A.opennem.org.au&oq=site%3A.opennem.org.au&aqs=chrome..69i57j69i58.4684j0j7&sourceid=chrome&ie=UTF-8), it evidently has not indexed the About Page, possibly because its not finding it. 

4. You don't have a [sitemap.xml](https://opennem.org.au/sitemap.xml) file.  Well you do, but its not working.  

Put together, I think these things are adversely impacting your search ranking.   Google for whatever reason isn't finding the About Page, and there's no `XML Sitemap` to explicitly tell it.  As such, the one page of the site with the most google juice (e.g your About Page) is being wasted.

## Potential Solutions

If your care about search engine ranking , you really need to get the [About page](https://opennem.org.au/#/about) indexed with google, as that's your only content page.  

You've got a few options

* One way to do this is by explicitly telling Google via their  [Google Search Console](https://search.google.com/search-console/about) tool.   Before you can play with that, you need to get your site verified.   I can help with that if you want. 

* Alternatively you can get your Sitemap.xml page fixed, but you probably want to setup Google Search Console. 

* Even more simply than that you may wish to update the HTML markup to include an explicit link to the About Page.

There's lots of other things to be done here (mainly around writing distinct copy for each distinct page with a distinct URL) but for now you want to focus on getting the content you do have indexed.

**Comment by charliedotau** [comment]
Hi there,

I wanted to follow this up.    The steps taken since I first raised this issue - creating an `XML sitemap` and submitting the sitemap to Google via `Google Search Console (GSC)` has not had the desired effect.  Google is still not indexing any page other than the root URL.

## Discovered - not currently indexed
Firstly, I can see in GSC that whilst Google has discovered a number of pages on the site (thanks to the XML Sitemap) it has not indexed them.  you can see this easily [via this link](https://www.google.com.au/search?q=site%3Aopennem.org.au&oq=site%3Aopennem.org.au&aqs=chrome..69i57j69i58.4623j0j7&sourceid=chrome&ie=UTF-8).  The unindexed pages are reported in GSC as `Discovered - not currently indexed`.  It doesn't explicitly say why.

## Fetch
Secondly, when I use the [Fetch as Google tool](https://www.google.com/webmasters/tools/googlebot-fetch?hl=en&siteUrl=https://opennem.org.au/) and try to fetch the `About page`, the tool returns the HTML it fetches.   This gives us insight as to how the Google Bot sees our content.  If you inspect the HTML the Fetch tool returns,  its quite revealing.   None of the content that a human sees in the About page, is in the markup.  

I guess this simply reflects how the current VueJS handles the content.

## Screaming Frog

Likewise, when I crawl the site using the wonderful [Screaming Frog](https://www.screamingfrog.co.uk/seo-spider/) crawler, it only finds one page.

## We has a problem

Put together, I conclude the site ain't indexable in its current form. 

## Why this matters

Organic (unpaid) Search traffic is the gift that keeps on giving: its free, targeted traffic.   But the site won't rank without copy.  And if the About page can't be crawled, there is no copy.   

We need to find a way to resolve this - be it with `Nuxtjs` or whatever - in order to grow the reach of the site cost effectively. 

Thoughts @chienleng ?

**Comment by chienleng** [comment]
Agreed about making some of the pages more statically available where it makes sense. Moving to Nuxt.js will be logical as it's still within the same Vue framework and I am already looking into how to best transitioned across.

**Comment by aleith** [comment]
Came across the concept of Universal / Isomorphic Applications which applies here:

> Although SPAs provide a great user experience, depending on your needs, they might not be the optimal solution — especially if you need quicker initial response times or optimal indexing by search engines.
> 
> There’s a fairly recent approach to solving these problems, called Isomorphic (or Universal) JavaScript applications. In this type of architecture, most of the code can be executed both on the server and the client. You can choose what you want to render on the server for a faster initial page load, and after that, the client takes over the rendering while the user is interacting with the app. _Because pages are initially rendered on the server, search engines can index them properly_.
